Code/Framework/AzFramework/Platform/Common/Xcb/azframework_xcb_files.cmake

```diff
@@ -6,6 +6,43 @@
 #
 #
 
+# If not set explicitly, look for a splash image in the project directory
+if(NOT DEFINED SPLASH_FILE)
+    # start by looking for file in the project directory
+    if(EXISTS ${PROJECT_SOURCE_DIR}/Resources/Splash.png)
+        set(SPLASH_FILE ${PROJECT_SOURCE_DIR}/Resources/Splash.png)
+    endif()
+    # if not found, assume it's engine centric build
+    if(DEFINED LY_PROJECTS)
+        if(EXISTS ${LY_PROJECTS}/Resources/Splash.png)
+            set(SPLASH_FILE ${LY_PROJECTS}/Resources/Splash.png)
+        endif()
+    endif ()
+endif()
+# If value is set, check if it exists and generate the header file
+if(DEFINED SPLASH_FILE)
+    file(TO_CMAKE_PATH ${SPLASH_FILE} SPLASH_FILE_RESOLVED_PATH)
+    # check if it's a png file
+    if(NOT ${SPLASH_FILE_RESOLVED_PATH} MATCHES ".*\\.png")
+        message(WARNING "Splash file must be a PNG image, omitting splash screen generation.")
+    else()
+        if(EXISTS ${SPLASH_FILE_RESOLVED_PATH})
+            execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/AzFramework)
+            execute_process(COMMAND ${CMAKE_COMMAND} -E copy ${SPLASH_FILE_RESOLVED_PATH} ${CMAKE_BINARY_DIR}/AzFramework/Splash.png)
+            file(REMOVE ${CMAKE_BINARY_DIR}/AzFramework/Splash.h)
+            add_definitions(-DSPLASH_IMAGE_EXISTS)
+            execute_process(COMMAND xxd -i Splash.png Splash.h
+                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/AzFramework
+            )
+            include_directories(${CMAKE_BINARY_DIR}/AzFramework)
+        else ()
+            message(WARNING "${SPLASH_FILE} not found, no splash screen will be displayed"
+                    " (set SPLASH_FILE to a valid path to a splash image to display it)")
+        endif()
+    endif()
+endif()
+
+
```

Comment:
This is not the right place to put this logic. This needs to be in a separate cmake file. Then the separate cmake file can be included in the "xcb" if block in `platform_nativeui_linux.cmake`